### PR TITLE
Expand tilde in neomake#utils#ExpandArgs()

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/neomake/neomake/issues/1089

### CODE REVIEW

File: autoload/neomake/utils.vim [hunk 1]
```diff
@@ -379,8 +379,14 @@ function! neomake#utils#ExpandArgs(args) abort
     let isk = &iskeyword
     set iskeyword=p,h,t,r,e,%,:
     try
+        let home = expand('~')
         let ret = map(a:args,
                     \ 'substitute(v:val, '
+                    \ . '''\(\%(\\\@<!\\\)\@<!\~\)'', '
```
Comment (tweekmonster) [inline autoload/neomake/utils.vim]: I tried to include this in the pattern below, but I was having a hard time getting it to work reliably.

**Comment by tweekmonster** [comment]
There appears to be a problem with the `Stringify()` test.  I thought that my changes were causing it to fail, but it seems to fail on master with `make docker_test`.

**Comment by blueyed** [comment]
> There appears to be a problem with the `Stringify()` test.  I thought that my changes were causing it to fail, but it seems to fail on master with `make docker_test`.

Works for me.  How does it fail?
I've noticed that it fails for nvim 0.1.7 though:

>    (108/147) [EXECUTE] (X) '{f2: function(''8'', {…})}' should be equal to '{f2: function(''8'')}'
> tests/utils.vader:311: AssertEqual neomake#utils#Stringify(obj), "{f2: function('".(fnr_base + 1)."', {…})}"

Any idea what patch (that was added later in Neovim) is that?
Otherwise I would just assume the new behavior with neovim 0.2.0?!

**Comment by tweekmonster** [comment]
Sorry, I should've posted the error.  I get this from the Vim 8.0.0238 test:

```
(108/149) [EXECUTE] (X) Vim(let):E704: Funcref variable name must start with a capital: exp
```

But, I don't see `exp` anywhere.

File: autoload/neomake/utils.vim [hunk 2]
```diff
@@ -379,16 +379,15 @@ function! neomake#utils#ExpandArgs(args) abort
...
-                    \ . 'home, '
+                    \ . '''\(\%(\\\@<!\\\)\@<!%\%(%\|\%(:[phtre]\+\)*\)\ze\)\w\@!'', '
+                    \ . '''\=(submatch(1) == "%%" ? "%" : expand(submatch(1)))'', '
                     \ . '''g'')')
         let ret = map(ret,
                     \ 'substitute(v:val, '
-                    \ . '''\(\%(\\\@<!\\\)\@<!%\%(%\|\%(:[phtre]\+\)*\)\ze\)\w\@!'', '
-                    \ . '''\=(submatch(1) == "%%" ? "%" : expand(submatch(1)))'', '
+                    \ . '''\(\%(\\\@<!\\\)\@<!\~\)'', '
+                    \ . 'expand(''~''), '
```
Comment (tweekmonster) [inline autoload/neomake/utils.vim]: Why did you switch the order?  This doesn't bother me.  I'm just curious.

Comment (blueyed) [inline autoload/neomake/utils.vim]: I thought it would be better, in case $HOME expands to something that would get picked up later then.

**Comment by blueyed** [comment]
You need to update Vader (https://github.com/junegunn/vader.vim/pull/121).

**Comment by tweekmonster** [comment]
@blueyed Forgot to mention it last night.  I cloned your fork of vader in the test directory and the test passed.  A symlink outside of the repo was causing the docker test to hang.

**Comment by blueyed** [comment]
FWIW: I am using a symlink to use my Vader always: ./tests/vim/plugins/vader.override -> /home/user/.dotfiles/vim/plugged/vader.vim/
The issue itself should be fixed in Vader itself, but the once checked out version is not updated automatically.
Are you using https://github.com/junegunn/vader.vim/pull/107 then now?

**Pull request merged**